pinot-core/src/main/java/org/apache/pinot/core/query/scheduler/QueryScheduler.java

```diff
@@ -103,7 +115,60 @@ public QueryScheduler(PinotConfiguration config, QueryExecutor queryExecutor, Re
    * @return Listenable future for query result representing serialized response. It is possible that the
    *    future may return immediately or be scheduled for execution at a later time.
    */
-  public abstract ListenableFuture<byte[]> submit(ServerQueryRequest queryRequest);
+  public ListenableFuture<byte[]> submit(ServerQueryRequest queryRequest) {
+    ListenableFuture<byte[]> future = submitInternal(queryRequest);
+    if (_enableQueryCancellation) {
+      String queryId = queryRequest.getQueryId();
+      // Track the running query for cancellation.
+      LOGGER.debug("Keep track of running query: {}", queryId);
```

Comment:
Suggest checking the log level in a if statement instead of directly logging the log here.